README.md

```diff
@@ -1,52 +1,205 @@
+
+Making all defaults explicit makes it possible to prevent unsafe changes when a newer version of the operator changes a default value.
+
+**Some values must be carried forward from the previous configuration.**
+
+Some values are computed at run-time but can never be changed. For example, the MTU of the overlay network is determined from the node's interfaces. Changing this is unsafe, so this must always be carried forward.
+
+Note that the fully-expressed configuration is not persisted back to the apiserver. Instead, it is saved only in the stored applied configuration. An alternative would be to inject these values via a mutating webhook. That requires a running service network, which we don't have until after we've run.
+
+### Validation
+Each network provider is responsible for validating their view of the configuration. For example, the OpenShiftSDN provider validates that the vxlanPort is a number between 1 and 65535, that the MTU is sane, etc. No validation is provided via the CRD itself.
+
+## Building
+
+### Test builds
+Build binaries manually with
+```
+./hack/build-go.sh
+```
+
+There is a special mode, the renderer, that emulates what the operator _would_ apply, given a fresh cluster. You can execute it manually:
+
+```
+_output/linux/amd64/cluster-network-renderer --config sample-config.yaml --out out.yaml
+```
+
+## Running manually against a test cluster
+If you have a running cluster, you can run the operator manually. Just set the `KUBECONFIG` environment variable.
+
+Note that, like all operators installed by the Cluster Version Operator, it determines all pointers to dependent images via environment variables. You will have to set several environment varables to emulate how the CVO works.
+
+You can determine the current values of these environment variables by inspecting the valid deployed daemonset. Do this before you delete it...
+```sh
+kubectl get -n openshift-cluster-network-operator daemonset cluster-network-operator -ojson | jq -Mrj '.spec.template.spec.containers[0].env[] | select(has("value")) | @text "\(.name)=\(.value) "' | tee images
+```
+
+After stopping the deployed operator (see below), you can run the operator locally with
+```sh
+env POD_NAME=LOCAL $(cat images) _output/linux/amd64/cluster-network-operator
+```
+
+### Stopping the deployed operators
+If the installer-deployed operator is up and running thanks to the CVO, you will need to stop the CVO, then stop the operator. If you don't stop the CVO, it will quickly re-create the production network operator daemonset.
+
+To do this, just scale the CVO down to 0 replicas and delete the network-operator daemonset.
+
+```sh
+kubectl scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
+kubectl delete -n openshift-cluster-network-operator daemonset cluster-network-operator 
+```
+
+## Building images
+By default, podman is used to build images. 
```

Comment:
Are the images pushed to some repo as part of the build? Where do we tell the daemonsets which image to use?

Reply:
Good question; I'm not sure that documenting the OpenShift build process is in-scope for this document. If you want to, go for it!